File: Editorial/0137_Single_Number_II/C/3_Mathematics.c
struct hash {
    int key;
    UT_hash_handle hh;
};
int singleNumber(int* nums, int numsSize) {
    struct hash *unique = NULL, *s;
    long sum = 0, uniqueSum = 0;
    for (int i = 0; i < numsSize; i++) {
        sum += nums[i];
        HASH_FIND_INT(unique, &nums[i], s);
        if (s == NULL) {
            s = malloc(sizeof(struct hash));
            s->key = nums[i];
            HASH_ADD_INT(unique, key, s);
            uniqueSum += nums[i];
        }
    }
    return (uniqueSum * 3 - sum) / 2;
}

File: Editorial/137_Single_Number_II/Go/3_Mathematics.go
func singleNumber(nums []int) int {
    unique := map[int]bool{}
    var sum, uniqueSum int
    for _, num := range nums {
        if _, exist := unique[num]; !exist {
            unique[num] = true
            uniqueSum += num
        }
        sum += num
    }
    return (3*uniqueSum - sum) / 2
}
